src/app/pages/home/exchange/exchange.page.ts

```diff
@@ -1,26 +1,157 @@
+
+          // Fix for ultra small values
+          if (Math.abs(translateX) < 0.001) translateX = 0;
+          if (Math.abs(translateY) < 0.001) translateY = 0;
+          if (Math.abs(translateZ) < 0.001) translateZ = 0;
+          if (Math.abs(rotateY) < 0.001) rotateY = 0;
+          if (Math.abs(rotateX) < 0.001) rotateX = 0;
+
+          const slideTransform = `translate3d(${translateX}px,${translateY}px,${translateZ}px)  rotateX(${rotateX}deg) rotateY(${rotateY}deg)`;
+
+          $slideEl.transform(slideTransform);
+          $slideEl[0].style.zIndex = -Math.abs(Math.round(offsetMultiplier)) + 1;
+          if (params.slideShadows) {
+          // Set shadows
+          let $shadowBeforeEl = isHorizontal ? $slideEl.find('.swiper-slide-shadow-left') : $slideEl.find('.swiper-slide-shadow-top');
+          let $shadowAfterEl = isHorizontal ? $slideEl.find('.swiper-slide-shadow-right') : $slideEl.find('.swiper-slide-shadow-bottom');
+          if ($shadowBeforeEl.length === 0) {
+              $shadowBeforeEl = swiper.$(`<div class="swiper-slide-shadow-${isHorizontal ? 'left' : 'top'}"></div>`);
+              $slideEl.append($shadowBeforeEl);
+          }
+          if ($shadowAfterEl.length === 0) {
+              $shadowAfterEl = swiper.$(`<div class="swiper-slide-shadow-${isHorizontal ? 'right' : 'bottom'}"></div>`);
+              $slideEl.append($shadowAfterEl);
+          }
+          if ($shadowBeforeEl.length) $shadowBeforeEl[0].style.opacity = offsetMultiplier > 0 ? offsetMultiplier : 0;
+          if ($shadowAfterEl.length) $shadowAfterEl[0].style.opacity = (-offsetMultiplier) > 0 ? -offsetMultiplier : 0;
+          }
+        }
+
+        // Set correct perspective for IE10
+        if (swiper.support.pointerEvents || swiper.support.prefixedPointerEvents) {
+           const ws = $wrapperEl[0].style;
+           ws.perspectiveOrigin = `${center}px 50%`;
+        }
+      },
+      setTransition(duration) {
+        const swiper = this;
+        swiper.slides
+          .transition(duration)
+          .find('.swiper-slide-shadow-top, .swiper-slide-shadow-right, .swiper-slide-shadow-bottom, .swiper-slide-shadow-left')
+          .transition(duration);
       }
-    };
-    this.router.navigate(['link'], navigationExtras);
+    }
+  }
+
+  constructor(
+    private authService: AuthService,
+    private userService: UserService,
+    private cardService: CardService,
+    private route: ActivatedRoute,
+    private router: Router,
+    public actionSheetController: ActionSheetController,
+    public alertController: AlertController,
+    private toastCtrl: ToastController,
+    public modalController: ModalController
+  ) {
+    console.log("Exchange page Start");
+    console.log('Is authenticated:', this.authService.isAuthenticated());
   }
 
   ngOnInit() {
+    console.log('Exchange Page Init');
+    this.noCardFound = true;
+    this.id = this.route.snapshot.paramMap.get('id');
+
+    if (this.id) {
```

Comment:
If `this.id` exists then `this.noCardFound` should be false. Also, I'd recommend naming the field positively (i.e. `this.isCardFound`) as that is the standard boolean naming convention.